geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/serial/SerialGatewaySenderAlterOperationsDUnitTest.java

```diff
@@ -0,0 +1,885 @@
+        Set<RegionQueue> queues = concurrentEventProcessor.getQueues();
+        for (RegionQueue queue : queues) {
+          if (queue instanceof SerialGatewaySenderQueue) {
+            assertThat(((SerialGatewaySenderQueue) queue).isRemovalThreadAlive())
+                .isFalse();
+          }
+        }
+      }
+    }
+  }
+
+  private int createReceiver() throws IOException {
+    int receiverPort = getRandomAvailableTCPPort();
+
+    GatewayReceiverFactory gatewayReceiverFactory = getCache().createGatewayReceiverFactory();
+    gatewayReceiverFactory.setStartPort(receiverPort);
+    gatewayReceiverFactory.setEndPort(receiverPort);
+    gatewayReceiverFactory.setManualStart(true);
+
+    GatewayReceiver receiver = gatewayReceiverFactory.create();
+    receiver.start();
+
+    return receiverPort;
+  }
+
+  private void validateRegionSize(String regionName, int regionSize) {
+    try (IgnoredException ie1 = addIgnoredException(CacheClosedException.class);
+        IgnoredException ie2 = addIgnoredException(ForceReattemptException.class)) {
+      Region region = getCache().getRegion(SEPARATOR + regionName);
+
+      await()
+          .untilAsserted(() -> {
+            assertThat(region.keySet()).hasSize(regionSize);
+          });
+    }
+  }
+
+  private void validateQueueContents(String senderId, int regionSize) {
+    try (IgnoredException ie1 = addIgnoredException(GatewaySenderException.class);
+        IgnoredException ie2 = addIgnoredException(InterruptedException.class)) {
+      InternalGatewaySender sender = findInternalGatewaySender(senderId);
+
+      if (sender.isParallel()) {
+        RegionQueue regionQueue = sender.getQueues().toArray(new RegionQueue[1])[0];
+        await()
+            .untilAsserted(() -> {
+              assertThat(regionQueue.size())
+                  .isEqualTo(regionSize);
+            });
+
+      } else {
+        Set<RegionQueue> queues = sender.getQueues();
+        await()
+            .untilAsserted(() -> {
+              int size = 0;
+              for (RegionQueue queue : queues) {
+                size += queue.size();
+              }
+              assertThat(size)
+                  .isEqualTo(regionSize);
+            });
+      }
+    }
+  }
+
+  private void validateSenderPausedState(String senderId) {
+    GatewaySender sender = findGatewaySender(senderId);
+
+    assertThat(sender.isPaused())
+        .isTrue();
+  }
+
+  private void validateSenderResumedState(String senderId) {
+    GatewaySender sender = findGatewaySender(senderId);
+
+    assertThat(sender.isPaused())
+        .isFalse();
+    assertThat(sender.isRunning())
+        .isTrue();
+  }
+
+  private void updateBatchSize(int batchsize) {
+    vm4.invoke(() -> {
+      AbstractGatewaySender sender = (AbstractGatewaySender) cache.getGatewaySender("ln");
+      boolean paused = false;
+      if (sender.isRunning() && !sender.isPaused()) {
+        sender.pause();
+        paused = true;
```

Comment:
Same comment as before on the ParallelGateway sender tests. If it is necessary that the gateway sender is paused while parameters are changed, that would have to be documented. Otherwise, I would remove the pause here and in the other parameter changes, below.